### PR TITLE
Add rb_range_beg_len

### PR DESCRIPTION
**Pull request by bjfish**
Used in `nokogiri`.

### CODE REVIEW

File: src/main/c/cext/ruby.c [hunk 1]
```diff
@@ -2263,6 +2263,48 @@ VALUE rb_range_new(VALUE beg, VALUE end, int exclude_end) {
...
+    if (len < 0)
+	len = 0;
+
+    *begp = beg;
+    *lenp = len;
+    return Qtrue;
+
+  out_of_range:
+    if (err) {
+	rb_raise(rb_eRangeError, "%d..%s%d out of range",
```
Comment (bjfish) [inline src/main/c/cext/ruby.c]: the original format string here didn't work `"%ld..%s%ld out of range"`

Comment (eregon) [inline src/main/c/cext/ruby.c]: `%ld` would be correct though for `long`. Was was the error?

File: src/main/c/cext/ruby.c [hunk 2]
```diff
@@ -2263,6 +2263,48 @@ VALUE rb_range_new(VALUE beg, VALUE end, int exclude_end) {
   return (VALUE) truffle_invoke(RUBY_CEXT, "rb_range_new", beg, end, exclude_end);
 }
 
+VALUE rb_range_beg_len(VALUE range, long *begp, long *lenp, long len, int err) {
```
Comment (eregon) [inline src/main/c/cext/ruby.c]: Is still simply copied from MRI?

File: src/main/c/cext/ruby.c (hunk 1 above)
Comment (bjfish) [inline src/main/c/cext/ruby.c]: Malformed format string %l

File: src/main/c/cext/ruby.c (hunk 2 above)
Comment (bjfish) [inline src/main/c/cext/ruby.c]: Yes

File: src/main/c/cext/ruby.c (hunk 1 above)
Comment (eregon) [inline src/main/c/cext/ruby.c]: Ah, that's a bug of our `rb_raise` implementation then.

**Comment by bjfish** [comment]
@chrisseaton I've rebased this and CI is running again.

**Pull request merged**